alacritty_terminal/src/term/mod.rs

```diff
@@ -830,13 +920,122 @@ impl SizeInfo {
+                        // Trim start/end with partially visible block selection
+                        (max(limit_start, start.col), min(limit_end, end.col))
+                    } else {
+                        // Multiple lines, so do full width damage
+                        (Column(0), cols - 1)
+                    };
+
+                    Some((start_col, Line(start.line), end_col, Line(end.line)))
+                }
+                None => None,
+            }
+            None => None,
+        }
+    }
+
+    /// Finalize and retrieve current damage. Note that the user is responsible
+    /// for calling LineDamage::reset on every line returned for efficiency.
+    pub fn get_damage(&mut self) -> (bool, &mut Vec<LineDamage>) {
+        self.damage.expand_line_damage(self.cursor.point.line, self.cursor.point.col, self.cursor.point.col);
+
+        let full = self.damage.full_damage ||
+            self.mode.contains(TermMode::INSERT);
+
+        if !full {
+            let cols = self.grid.num_cols();
+
+            //
+            // Selections
+            //
+            if let Some(sel_damage) = self.calculate_selection_damage(cols, &self.grid.selection) {
+                for line in (sel_damage.1).0..=(sel_damage.3).0 {
+                    self.damage.expand_line_damage(Line(line), sel_damage.0, sel_damage.2)
+                }
+            }
+            if let Some(sel_damage) = self.calculate_selection_damage(cols, &self.last_selection) {
+                for line in (sel_damage.1).0..=(sel_damage.3).0 {
+                    self.damage.expand_line_damage(Line(line), sel_damage.0, sel_damage.2)
+                }
+            }
+
+            //
+            // URL highlights
+            //
+            if let Some(hl_damage) = self.calculate_highlight_damage(cols, &self.grid.url_highlight) {
+                for line in (hl_damage.1).0..=(hl_damage.3).0 {
+                    self.damage.expand_line_damage(Line(line), hl_damage.0, hl_damage.2)
+                }
+            }
+            if let Some(hl_damage) = self.calculate_highlight_damage(cols, &self.last_highlight) {
+                for line in (hl_damage.1).0..=(hl_damage.3).0 {
+                    self.damage.expand_line_damage(Line(line), hl_damage.0, hl_damage.2)
+                }
+            }
+        }
+
+        self.last_selection = self.grid.selection.clone();
+        self.last_highlight = self.grid.url_highlight.clone();
+        self.damage.full_damage = false;
+        (full, &mut self.damage.line_damage)
+    }
+
+    /// Damage the current cursor position. Must be done after a call to
+    /// Term::get_damage.
+    pub fn damage_cursor(&mut self) {
+        self.damage.expand_line_damage(self.cursor.point.line, self.cursor.point.col, self.cursor.point.col);
+    }
+
+    /// Reset the current damage without reading it.
+    pub fn reset_damage(&mut self) {
+        self.damage.clear_damage();
+        self.damage_cursor();
```

Comment:
Why are we damaging cursor in reset? I don't think that its a `reset` job, right?

Reply:
If the current cursor position is not damaged after all damage state is cleared, then the damage tracker will have no idea what the crsor start position was in this damage run, resulting in defective damage reports.

But could just be the job of the caller. In get_damage, I am unable to do it for the caller, so consistency might be nice.

Reply:
Oh, that make sense, lets just add a comment on why this is happening.